pennylane/devices/default_qubit.py

```diff
@@ -736,7 +740,16 @@ def compute_vjp(
 
         max_workers = execution_config.device_options.get("max_workers", self._max_workers)
         if max_workers is None:
-            res = tuple(adjoint_vjp(circuit, cots) for circuit, cots in zip(circuits, cotangents))
+
+            def _state(circuit):
+                return (
+                    None if self._state_cache is None else self._state_cache.get(circuit.hash, None)
```

Comment:
Agreed, then you can just do this

```suggestion
                    self._state_cache.get(circuit.hash, None)
```